compiler/src/main/java/com/squareup/anvil/compiler/codegen/CodeGenerationExtension.kt

```diff
@@ -69,6 +69,8 @@ internal class CodeGenerationExtension(
     val anvilModule = RealAnvilModuleDescriptor(module)
     anvilModule.addFiles(files)
 
+    val nonPrivateCodeGenerators = codeGenerators.filter { it !is PrivateCodeGenerator }
```

Comment:
You could do something like:

```kotlin
val (privateCodeGenerators, nonPrivateCodeGenerators) =
  codeGenerators.partition { it is PrivateCodeGenerator }
```
to separate out the two kinds of code generators, and then use `privateCodeGenerators` instead of the extra `codeGenerators.filterIsInstance<PrivateCodeGenerator>()` pass below. Up to you.

Reply:
I had no idea, that's gorgeous.